sw/device/tests/entropy_src_fw_observe_many_contiguous.c

```diff
@@ -103,9 +108,68 @@ static status_t let_observe_fifo_overflow(uint32_t timeout_usec) {
   return OK_STATUS();
 }
 
+void ottf_external_isr(uint32_t *exc_info) {
+  dif_rv_plic_irq_id_t plic_irq_id;
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&rv_plic, kTopEarlgreyPlicTargetIbex0,
+                                     &plic_irq_id));
+
+  top_earlgrey_plic_peripheral_t peripheral =
+      top_earlgrey_plic_interrupt_for_peripheral[plic_irq_id];
+  CHECK(peripheral == kTopEarlgreyPlicPeripheralEntropySrc);
+
+  CHECK(plic_irq_id == kTopEarlgreyPlicIrqIdEntropySrcEsObserveFifoReady);
+
+  CHECK(!entropy_src_fifo_has_overflowed(),
+        "observe FIFO overflowed during collection");
+
+  uint32_t buf[32];
+  size_t len = ARRAYSIZE(buf);
+  CHECK_DIF_OK(
+      dif_entropy_src_observe_fifo_nonblocking_read(&entropy_src, buf, &len));
+
+  if (len > 0) {
+    size_t words_written = 0;
+    // This is a non-blocking write that should be safe to run in this IRQ.
+    CHECK_DIF_OK(dif_entropy_src_observe_fifo_write(&entropy_src, buf, len,
```

Comment:
Oh good, I did think this was odd. I'm happy to rename this DIF in this PR, what would make a better name?

`dif_entropy_src_packer_fifo_write`?

I think @pamaury also wanted `_nonblocking` adding to the name, is that correct?

Reply:
I think the name should mention that it is not blocking. Not sure about the name of the function, @h-filali do you have a suggestion?

Reply:
Yes it would be great if you could do that @jwnrt .
@pamaury probably something like `dif_entropy_src_fw_ov_data_write` since the register is called [`fw_ov_wr_data`](https://opentitan.org/book/hw/ip/entropy_src/doc/registers.html#fw_ov_wr_data).